PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -11,8 +11,31 @@
 import org.json.JSONObject;
 
 /**
- * Used to create and tokenize PayPal accounts. For more information see the
- * <a href="https://developers.braintreepayments.com/guides/paypal/overview/android/">documentation</a>
+ * PayPalClient enables you to obtain permission to charge your customers' PayPal accounts by presenting the PayPal website.
+ *
+ * To make PayPal available, you must ensure that PayPal is enabled in your Braintree control panel.
+ * See our [online documentation](https://developers.braintreepayments.com/guides/paypal/overview/android/) for
+ * details.
+ *
+ * This class supports two basic use-cases: Vault and Checkout. Each of these involves variations on the
+ * user experience as well as variations on the capabilities granted to you by this authorization.
+ *
+ * The *Vault* option uses PayPal's future payments authorization, which allows your merchant account to
+ * charge this customer arbitrary amounts for a long period of time into the future (unless the user
+ * manually revokes this permission in their PayPal control panel.) This authorization flow includes
+ * a screen with legal language that directs the user to agree to the terms of Future Payments.
+ * Unfortunately, it is not currently possible to collect shipping information in the Vault flow.
+ *
+ * The *Checkout* option creates a one-time use PayPal payment on your behalf. As a result, you must
+ * specify the checkout details up-front, so that they can be shown to the user during the PayPal flow.
+ * With this flow, you must specify the estimated transaction amount, and you can collect shipping
+ * details. This flow omits the Future Payments agreement, and the resulting payment method cannot be
+ * stored in the vault. It is only possible to create one Braintree transaction with this form of user
+ * approval.
+ *
+ * Upon successful completion, you will receive a PayPalAccountNonce, which includes user-facing
+ * details and a payment method nonce, which you must pass to your server in order to create a transaction
+ * or save the authorization in the Braintree vault (not possible with Checkout).
```

Comment:
I think it's a little confusing to say that you can't vault a payment method created with the Checkout flow, since you actually _can_ do this if you set `requestBillingAgreement` to `true` on the `PayPalCheckoutRequest`.

Reply:
These docs might just be outdated?

Reply:
I took this description from `BTPayPalDriver` since our `PayPalClient` doc string was fairly short and this seemed more descriptive, but if these are outdated I'll just put it back to the old doc string and maybe we can do another pass at updating doc strings to be more descriptive and accurate.

Reply:
Ahhhh. I'll make a note for us to update these docs on iOS as well.